File: 프로그래머스/lv1/최원준/로또의 최고 순위와 최저 순위.py
# https://www.acmicpc.net/problem/

'''
1. 아이디어 :

2. 시간복잡도 :
    O(
3. 자료구조 :

'''


def solution(lottos, win_nums):
    prize = {6: 1, 5: 2, 4: 3, 3: 4, 2: 5, 1: 6, 0: 6}
    win_nums = set(win_nums)
    zero = match = 0
    for n in lottos:
        zero = zero+1 if n == 0 else zero
        match = match+1 if n in win_nums else match

    return [prize[match + zero], prize[match]]

File: 프로그래머스/lv2/김서영2/로또의 최고순위와 최저순위.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    int match = 0, zeros = 0;
    for(int i=0; i<lottos.size(); i++) {
        if(lottos[i] == 0) zeros++;
        else {
            auto it = find(win_nums.begin(), win_nums.end(), lottos[i]);
            if(it != win_nums.end()) match++;
        }
    }

    int min_rank = 7 - match;
    int max_rank = 7 - (match + zeros);

    if(min_rank > 6) min_rank = 6;
    if(max_rank > 6) max_rank = 6;


    answer.push_back(max_rank);
    answer.push_back(min_rank);

    return answer;
}